test/fuzz/transformation_composite_insert_test.cpp

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_composite_insert.h"
+
+#include "source/fuzz/instruction_descriptor.h"
+#include "test/fuzz/fuzz_test_util.h"
+
+namespace spvtools {
+namespace fuzz {
+namespace {
+
+TEST(TransformationCompositeInsertTest, BasicScenarios) {
+  // This is a simple transformation and this test handles the main cases.
```

Comment:
Add tests for the following cases:
1. `composite_id` is invalid (points to nonexistent instruction or instruction without a type id)
2. `object_id` is invalid (same conditions as above).
3. You can add an ASSERT_DEATH test for the case when the inserted component is a pointer if you add a corresponding assertion into `IsApplicable` method.
4. The case when it's not possible to insert `OpCompositeInsert` before the instruction descriptor.
5. Check that applied transformations create required synonyms.
6. Check the case when either `composite_id` or `object_id` is irrelevant (synonyms shouldn't be created in that case).
7. The composite has no elements.

This test case will become huge and hard to understand if you attempt to add all of the situations into it. Thus, it might be better to create separate test cases for some of them.

Reply:
Could you give an example when it is not possible to insert OpCompositeInsert before the instruction descriptor?

Reply:
For example, when an instruction descriptor is an OpBranchConditional with OpSelectionMerge above it. Or when instruction descriptor is an OpPhi.

Reply:
The reason why we need to test these cases is that, even though you are applying your transformation correctly in the fuzzer pass, we want to be sure that it will never apply incorrectly in the shrinker.